#! /bin/sh /usr/share/dpatch/dpatch-run
## 43_check_http_large_pages_mleak.dpatch
## From: Ton Voon <ton.voon@opsera.com>
## Date: Fri, 26 Feb 2010 12:47:38 +0000
## Subject: [PATCH] Fix memory leak in check_http for large pages (Jimmy Bergman - #2957455)
## X-Git-Url: http://repo.or.cz/w/nagiosplugins.git?a=commitdiff_plain;h=6b782ebfd4832c1fe621556bcf894162b8caa8aa
##
## DP:  Fix memory leak in check_http for large pages

@DPATCH@

---
 plugins/check_http.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletions(-)

diff --git a/plugins/check_http.c b/plugins/check_http.c
index 0a4b12b..5cdf144 100644
--- a/plugins/check_http.c
+++ b/plugins/check_http.c
@@ -784,6 +784,7 @@ check_http (void)
   int i = 0;
   size_t pagesize = 0;
   char *full_page;
+  char *full_page_new;
   char *buf;
   char *pos;
   long microsec;
@@ -871,7 +872,9 @@ check_http (void)
   full_page = strdup("");
   while ((i = my_recv (buffer, MAX_INPUT_BUFFER-1)) > 0) {
     buffer[i] = '\0';
-    asprintf (&full_page, "%s%s", full_page, buffer);
+    asprintf (&full_page_new, "%s%s", full_page, buffer);
+    free (full_page);
+    full_page = full_page_new;
     pagesize += i;
 
                 if (no_body && document_headers_done (full_page)) {
-- 
1.6.3

